src/Core/Domain/Product/Image/QueryResult/Shop/ShopProductImages.php

```diff
@@ -0,0 +1,61 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Product\Image\QueryResult\Shop;
+
+class ShopProductImages
+{
+    /**
+     * @var ProductImageCollection
+     */
+    private $productImageCollection;
+
+    /**
+     * @var int
+     */
+    private $shopId;
+
+    public function __construct(int $shopId, ProductImageCollection $productImageCollection)
+    {
+        $this->shopId = $shopId;
+        $this->productImageCollection = $productImageCollection;
+    }
+
+    /**
+     * @return int
+     */
+    public function getShopId(): int
+    {
+        return $this->shopId;
+    }
+
+    public function getProductImageCollection(): ProductImageCollection
```

Comment:
```suggestion
    public function getProductImages(): ProductImageCollection
```
Just a suggestion but since the method is strictly typed with a collection class maybe the `Collection` suffix in the method name is redundant?

Reply:
(if the previous suggestion I made is accepted we wouldn't even need this getter method since the class itself would be a collection that we can directly iterate on)